tensorflow/core/kernels/dense_update_functor_gpu.cu.cc

```diff
@@ -56,11 +56,13 @@ struct DenseUpdate<GPUDevice, T, SUB> {
 #define DEFINE_GPU_KERNELS(T)                              \
   template struct functor::DenseUpdate<GPUDevice, T, ADD>; \
   template struct functor::DenseUpdate<GPUDevice, T, SUB>;
+TF_CALL_int32(DEFINE_GPU_KERNELS);
```

Comment:
You can wait for XQ to return in a week or so - and he'll probably say that we shouldn't add these.  I'm perfectly OK with scatter_nd and gather_nd getting GPU kernels for int32/int64, but not scatter/gather.

Reply:
Any updates?

Reply:
Still waiting for a review

Reply:
@zheng-xq 
I don't know when you would do gather or scatter on shapes and ranks, but I understand if the default option shouldn't be on GPU.
1. Neither uint32 nor int64 are supported. This PR adds int32 and int64. Furthermore, operations like tf.unique return int32/int64 as indices, instead of uint32/uint64.
2. Wouldn't new type imply explicit casting? 
3. Seems ok. Could it be an option of the op, instead? Something like gather_nd(params, indices, prefer_device=True/False, name)?

I've submitted the issue and the pull request as I need to do a few gather/scatter_nd operations on int64 tensors for blur/convolution operations on graphs. So, I fetch(gather) vertices on a graph by the index of the vertex and update values on a graph with scatter_nd operation. The structure of the graph is conditioned on the elements of the batch, so it makes sense to run it on the GPU.

Reply:
Adding josh11b@, the authority on this issue. 

Josh, could you comment on the current TF policy on real int32 GPU kernels?